bitbybit/src/bitfield.rs

```diff
@@ -7,10 +7,14 @@ use quote::{quote, ToTokens};
 use syn::__private::TokenStream2;
 use syn::{Attribute, Data, DeriveInput, GenericArgument, PathArguments, Type};
 
+/// In the code below, bools are considered to have 0 bits. This lets us distinguish them
+/// from u1
+const BITCOUNT_BOOL: usize = 0;
```

Comment:
This kinda introduces a sentinel value, but it's for keeping things simple, should be alright.

Reply:
Yeah I am not super happy with it. The alternative would have been an enum here - would have been cleaner typing (and less lying about number of bits) but it would have complicated things somewhat, so I settled for the simple and pragmatic thing for now